controllers/datadogmonitor/controller.go

```diff
@@ -114,7 +114,7 @@ func (r *Reconciler) internalReconcile(ctx context.Context, req reconcile.Reques
 		logger.V(1).Info("Monitor ID is not set; creating monitor in Datadog")
 		// If the monitor ID is 0, then it doesn't exist yet in Datadog. Create the monitor (only metric alerts)
 		switch string(instance.Spec.Type) {
-		case string(datadogapiclientv1.MONITORTYPE_METRIC_ALERT):
+		case string(datadogapiclientv1.MONITORTYPE_METRIC_ALERT), string(datadogapiclientv1.MONITORTYPE_QUERY_ALERT), string(datadogapiclientv1.MONITORTYPE_SERVICE_CHECK):
```

Comment:
maybe, we can create a function `IsValidMetricType(type DatadogMonitorType)` and replace switch case by a simple `if ... else` if we know that we only have on `case`.